server/command_upgrade.go

```diff
@@ -85,7 +86,6 @@ func (p *Plugin) runUpgradeCommand(args []string, extra *model.CommandArgs) (*mo
 	// Only change the license if a value was provided.
 	license := installToUpgrade.License
 	if newLicense != "" {
```

Comment:
I think this logic needs to be updated. If license is not false we need to handle TE as well now, right?

Reply:
Ah, yeah, I started making this change and then backed it out due to a comment on the provisioner PR, thinking that we had decided that should go in a separate PR. And in that process, I left around the deletion of line 88. But now I'm seeing how adding this logic is still useful here even when changing the image type isn't supported by the provisioner.